benchmarks/python/test_cross_entropy_loss.py

```diff
@@ -125,3 +127,118 @@ def fwd_call(inp):
     run_benchmark(benchmark, unary_bwd_torch, [outputs, grads, *inputs])
+
+
+# Simple test of F.cross_entropy without final reduction
+# thunder won't use nvFuser for this fusion, here we manually define the fusion
+def nvfuser_cross_entropy_fusion(
+    fd: FusionDefinition, batch_size: int, vocab_size: int
+) -> None:
+    """
+    NvFuser fusion definition for torch.nn.functional.cross_entropy(logits, labels, reduction='none')
+    1. Compute LSE (log-sum-exp) for each row
+    2. Access target logit directly: logits[i, labels[i]]
+    3. Compute loss: loss = lse - target_logit
+    """
+    # Input tensors
+    T0 = fd.define_tensor(
+        shape=[batch_size, vocab_size],
+        contiguity=[True, True],
+        dtype=DataType.BFloat16,
+        is_cpu=False,
+        stride_order=[1, 0],
+    )
+    T1 = fd.define_tensor(
+        shape=[batch_size],
+        contiguity=[True],
+        dtype=DataType.Int,
+        is_cpu=False,
+        stride_order=[0],
+    )
+    T2 = fd.ops.cast(T0, dtype=DataType.Float)
+    T3 = fd.ops.max(T2, dims=[1], keepdim=False, dtype=DataType.Null)
+    T4 = fd.ops.broadcast_in_dim(T3, shape=[batch_size, 1], broadcast_dims=[0])
+    T11 = fd.ops.broadcast_in_dim(
+        T4, shape=[batch_size, vocab_size], broadcast_dims=[0, 1]
+    )
+    T12 = fd.ops.sub(T2, T11)
+    T13 = fd.ops.exp(T12)
+    T14 = fd.ops.sum(T13, dims=[1], keepdim=False, dtype=DataType.Null)
+    T15 = fd.ops.log(T14)
+    T16 = fd.ops.add(T15, T3)
+    T17 = fd.ops.broadcast_in_dim(T1, shape=[batch_size, 1], broadcast_dims=[0])
+    T18 = fd.ops.gather(T0, T17, dim=1)
+    T19 = fd.ops.squeeze(T18, dims=[1])
+    T20 = fd.ops.cast(T19, dtype=DataType.Float)
+    T21 = fd.ops.sub(T16, T20)
+    T22 = fd.ops.cast(T21, dtype=DataType.BFloat16)
+    fd.add_output(T22)
+
+
+@pytest.mark.parametrize("vocab_size", SyntheticMiniModel.sizes_from_models)
+def test_function_cross_entropy_fwd_nvf_benchmark(
+    benchmark, vocab_size: int, disable_validation: bool, disable_benchmarking: bool
+):
+    batch_size = 4096
+    inputs = [
+        0.1
+        * torch.randn(
+            batch_size,
+            vocab_size,
+            device="cuda",
+            dtype=torch.bfloat16,
+            requires_grad=False,
+        ),
+        torch.randint(
+            0,
+            vocab_size,
+            (batch_size,),
+            device="cuda",
+            dtype=torch.int64,
+            requires_grad=False,
+        ),
+    ]
+    with FusionDefinition() as fd:
+        nvfuser_cross_entropy_fusion(fd, batch_size, vocab_size)
+
+    if not disable_validation:
+        fd.validate(inputs, [torch_cross_entropy_fwd(inputs)])
+
+    if not disable_benchmarking:
+        run_benchmark(benchmark, fd.execute, inputs)
+
+
+def quack_cross_entropy_fwd_wrapper(inputs: list):
+    return quack_cross_entropy_fwd(*inputs, return_dx=False)
+
+
+@pytest.mark.parametrize(
+    "executor", ["eager", "torchcompile", "thunder", "thunder-torchcompile", "quack"]
```

Comment:
How is Thunder going to schedule this then?

Reply:
cross entropy in thunder is split into `log_softmax` and `nll_loss` https://github.com/Lightning-AI/lightning-thunder/blob/dbad41c148c4220131efe8097ee0937b3d44bc5a/thunder/torch/__init__.py#L5165